AmplifyPlugins/API/AWSAPICategoryPlugin/Support/Decode/GraphQLResponseDecoder+DecodeData.swift

```diff
@@ -28,10 +28,98 @@ extension GraphQLResponseDecoder {
         } else {
             let serializedJSON = try encoder.encode(graphQLData)
             let responseData = try decoder.decode(request.responseType, from: serializedJSON)
+            if responseData is Model {
+                return try GraphQLResponseDecoder.decodeToModelWithArrayAssociations(responseType: request.responseType,
+                                                                                     modelGraphQLData: graphQLData)
+            } else if responseData is ModelListMarker {
+                return try decodeToAppSyncList(graphQLData: graphQLData)
+            }
             return responseData
         }
     }
 
+    static func decodeToModelWithArrayAssociations<M: Decodable>(responseType: M.Type,
+                                                                 modelGraphQLData: JSONValue) throws -> M {
+        let encoder = JSONEncoder()
+        encoder.dateEncodingStrategy = ModelDateFormatting.encodingStrategy
+        let decoder = JSONDecoder()
+        decoder.dateDecodingStrategy = ModelDateFormatting.decodingStrategy
+
+        guard let modelType = getModelType(modelGraphQLData: modelGraphQLData),
+              let id = getId(modelGraphQLData: modelGraphQLData),
+              case .object(var model) = modelGraphQLData else {
+            let serializedJSON = try encoder.encode(modelGraphQLData)
+            return try decoder.decode(M.self, from: serializedJSON)
+        }
+
+        // Iterate over the associations of the model and for each association, store it's association data
+        // For example, if the modelType is a Post and has a field that is an array association like Comment
+        // Store the post's id and post field in the comments as the `associationPayload`
+        modelType.schema.fields.values.forEach { modelField in
+            if modelField.isArray && modelField.hasAssociation,
+               let associatedField = modelField.associatedField {
+                let modelFieldName = modelField.name
+                let associatedFieldName = associatedField.name
+
+                if model[modelFieldName] == nil {
```

Comment:
I think I'm missing how this routine sets both the "id" and the actual payload, as you describe in the comment above. Can you clarify what this is doing?

Reply:
updated comment to clarify that we're not storing any actual payload. just the identifier and the name of the parent